publisher.go

```diff
@@ -0,0 +1,178 @@
+package main
+
+import (
+	"fmt"
+	"time"
+	"net/http"
+	"io/ioutil"
+	"log"
+	"github.com/jlaffaye/ftp"
+	"bytes"
+	"sync"
+	"io"
+)
+
+var lock = &sync.Mutex{}
```

Comment:
I didn't run into that issue when using a pure-ftpd server (though I did hit "max users" issues) so maybe it's down to ftp server implementation.